### ISSUE TITLE
Changing type of structure data variable in linear view is confusing

### ISSUE THREAD

**Issue by D0ntPanic** [Vector35/binaryninja-api#1793]
In linear view, if a data variable is declared as a structure like this:

```
00000c18  struct struct_1 data_c18 = 
00000c18  {
00000c18      int32_t field_0 = 0x148
00000c1c      int32_t field_4 = -0x45f
00000c20  }
```

If you select the first line and press 'Y', it would be expected that it would change the type of the variable. However, it instead tries to change the type of the first member in the structure. It should only change the first member's type when the member is selected.

It is possible to work around this issue by selecting the entire structure, then pressing 'Y'

**Comment by ccarpenter04** [comment]
If this is is actually the current behavior than I agree that it would be extremely confusing and not the behavior I would expect.

**Comment by jeffli678** [comment]
The current code is using the address of the selected line to do the work, and since the structure definition line has the same address as the first member, so the behavior is faulty.

Moreover, this actually reveals a larger frustration than the reported behavior. If one were to redefine a union, it is quite chaotic. Nested structures should also pose problems. 

The good news is, I have devised a way to disambiguate these lines as part of the effort to support "Display As" in linear view. Which should also help here.

**Comment by withzombies** [comment]
I just ran into this also. It's very confusing.

**Comment by jeffli678** [comment]
Fixed in dev 2.4.3012
